lib/galaxy/webapps/galaxy/api/users.py

```diff
@@ -730,6 +731,19 @@ def send_activation_email(
         if not self.service.user_manager.send_activation_email(trans, user.email, user.username):
             raise exceptions.MessageException("Unable to send activation email.")
 
+    @router.get(
+        "/api/users/{user_id}/roles",
+        name="get user roles",
+        description="Return a collection of roles associated with this user. Only admins can see user roles.",
+        require_admin=True,
```

Comment:
> ...what prompted the addition...

The endpoint was added to support API tests; so that a populator can still find a user's private role: 98044ad7dce0573fa6646adc29adcbbea8bf0e42

Previously, we retrieved all roles, and then did `[r for r in roles if r["name"] == user_email]`, where `roles` was a list of `RoleModelResponse` objects, representing *all* roles.

Now that the role name/user email comparison does not work, we do this instead: `private_roles = [r for r in roles if r["type"] == PRIVATE_ROLE_TYPE]`, where `roles` are the roles associated with that particular user.

> Why can't you see your own roles ? That would seem fine. 

No specific reason; I think I chose this by default simply because we don't use this other than for API tests? If we ever need this endpoint, we can change it. Also, if we were to remove the admin requirement, we'd have to add logic that a user can only view their roles, and not some other user roles.